pkg/controllermanager/apis/config/v1alpha1/defaults.go

```diff
@@ -18,331 +18,315 @@ import (
+	if obj.ResourceLock == "" {
+		// Don't use a constant from the client-go resourcelock package here (resourcelock is not an API package, pulls
+		// in some other dependencies and is thereby not suitable to be used in this API package).
+		obj.ResourceLock = "leases"
 	}
 
-	if obj.Controllers.ControllerRegistration == nil {
-		obj.Controllers.ControllerRegistration = &ControllerRegistrationControllerConfiguration{}
+	componentbaseconfigv1alpha1.RecommendedDefaultLeaderElectionConfiguration(obj)
+
+	if obj.ResourceNamespace == "" {
+		obj.ResourceNamespace = ControllerManagerDefaultLockObjectNamespace
 	}
-	if obj.Controllers.ControllerRegistration.ConcurrentSyncs == nil {
-		v := DefaultControllerConcurrentSyncs
-		obj.Controllers.ControllerRegistration.ConcurrentSyncs = &v
+	if obj.ResourceName == "" {
+		obj.ResourceName = ControllerManagerDefaultLockObjectName
 	}
+}
 
-	if obj.Controllers.ExposureClass == nil {
-		obj.Controllers.ExposureClass = &ExposureClassControllerConfiguration{}
+// SetDefaults_ShootRetryControllerConfiguration sets defaults for the ShootRetryControllerConfiguration.
+func SetDefaults_ShootRetryControllerConfiguration(obj *ShootRetryControllerConfiguration) {
+	if obj.ConcurrentSyncs == nil {
+		obj.ConcurrentSyncs = pointer.Int(DefaultControllerConcurrentSyncs)
+	}
+	if obj.RetryPeriod == nil {
+		obj.RetryPeriod = &metav1.Duration{Duration: 10 * time.Minute}
 	}
-	if obj.Controllers.ExposureClass.ConcurrentSyncs == nil {
-		v := DefaultControllerConcurrentSyncs
-		obj.Controllers.ExposureClass.ConcurrentSyncs = &v
+	if obj.RetryJitterPeriod == nil {
+		obj.RetryJitterPeriod = &metav1.Duration{Duration: 5 * time.Minute}
 	}
+}
 
-	if obj.Controllers.Project == nil {
-		obj.Controllers.Project = &ProjectControllerConfiguration{}
+// SetDefaults_SeedControllerConfiguration sets defaults for the given SeedControllerConfiguration.
+func SetDefaults_SeedControllerConfiguration(obj *SeedControllerConfiguration) {
+	if obj.SyncPeriod == nil {
+		obj.SyncPeriod = &metav1.Duration{Duration: 10 * time.Second}
 	}
-	if obj.Controllers.Project.ConcurrentSyncs == nil {
-		v := DefaultControllerConcurrentSyncs
-		obj.Controllers.Project.ConcurrentSyncs = &v
+	if obj.ConcurrentSyncs == nil {
+		obj.ConcurrentSyncs = pointer.Int(DefaultControllerConcurrentSyncs)
 	}
-	if obj.Controllers.Project.MinimumLifetimeDays == nil {
-		v := 30
-		obj.Controllers.Project.MinimumLifetimeDays = &v
+	if obj.MonitorPeriod == nil {
+		obj.MonitorPeriod = &metav1.Duration{Duration: 40 * time.Second}
 	}
-	if obj.Controllers.Project.StaleGracePeriodDays == nil {
-		v := 14
-		obj.Controllers.Project.StaleGracePeriodDays = &v
+	if obj.ShootMonitorPeriod == nil {
+		obj.ShootMonitorPeriod = &metav1.Duration{Duration: 5 * obj.MonitorPeriod.Duration}
 	}
-	if obj.Controllers.Project.StaleExpirationTimeDays == nil {
-		v := 90
-		obj.Controllers.Project.StaleExpirationTimeDays = &v
+}
+
```

Comment:
```suggestion

// SetDefaults_ProjectControllerConfiguration sets defaults for the ProjectControllerConfiguration.
```
Same for other functions as well.